### PR TITLE
feat(`participation`): initialize module

### LINKED ISSUE #569: Initialize the `participation` module
Initialize the `participation` module with `fundraising` and `staking` as dependencies
Implement all test utilities related to the module (module for test purpose). For this reason, I think we should first work on https://github.com/tendermint/spn/issues/498 as a priority since adding a new module would increase complexity of this package a lot

We should update imported `fundraising` module dep as well

### PR DESCRIPTION
**Pull request by giunatale**
Closes #569 

### What does this PR does?
Initialize module following requirements defined in #569.
Params are sketched although default values, test sampling and paramset validation are left as todos.

### CODE REVIEW

File: app/app.go [hunk 1]
```diff
@@ -256,6 +261,8 @@ type App struct {
 	CampaignKeeper    campaignmodulekeeper.Keeper
 	MonitoringcKeeper monitoringcmodulekeeper.Keeper
 	RewardKeeper      rewardmodulekeeper.Keeper
+
```
Comment (lubtd) [inline app/app.go:264]:
```suggestion
```

File: app/app.go [hunk 2]
```diff
@@ -477,6 +485,17 @@ func New(
 	)
 	monitoringcModule := monitoringcmodule.NewAppModule(appCodec, app.MonitoringcKeeper, app.AuthKeeper, app.BankKeeper)
 
+	app.ParticipationKeeper = *participationmodulekeeper.NewKeeper(
+		appCodec,
+		keys[participationmoduletypes.StoreKey],
+		keys[participationmoduletypes.MemStoreKey],
+		app.GetSubspace(participationmoduletypes.ModuleName),
+
```
Comment (lubtd) [inline app/app.go:493]:
```suggestion
```

File: testutil/keeper/keeper.go [hunk 3]
```diff
@@ -43,15 +46,17 @@ type TestKeepers struct {
 	MonitoringConsumerKeeper *monitoringckeeper.Keeper
 	BankKeeper               bankkeeper.Keeper
 	IBCKeeper                *ibckeeper.Keeper
+	ParticipationKeeper      *participationkeeper.Keeper
```
Comment (lubtd) [inline testutil/keeper/keeper.go:49]: We can also add `fundraising`, we will need it for `participation` keeper test

Comment (lubtd) [reply testutil/keeper/keeper.go:49]: Should not be necessary in `TestMsgServers` though

File: x/participation/keeper/keeper.go [hunk 4]
```diff
@@ -0,0 +1,51 @@
+package keeper
+
+import (
+	"fmt"
+
+	"github.com/tendermint/tendermint/libs/log"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	paramtypes "github.com/cosmos/cosmos-sdk/x/params/types"
```
Comment (lubtd) [inline x/participation/keeper/keeper.go:10]:
```suggestion
	"github.com/cosmos/cosmos-sdk/codec"
	sdk "github.com/cosmos/cosmos-sdk/types"
	paramtypes "github.com/cosmos/cosmos-sdk/x/params/types"
	"github.com/tendermint/tendermint/libs/log"
```

File: x/participation/keeper/keeper.go [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+		stakingKeeper     types.StakingKeeper
+	}
+)
+
+func NewKeeper(
+	cdc codec.BinaryCodec,
+	storeKey,
+	memKey sdk.StoreKey,
+	ps paramtypes.Subspace,
+
```
Comment (lubtd) [inline x/participation/keeper/keeper.go:31]:
```suggestion
```

File: x/participation/keeper/keeper.go [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+	fundraisingKeeper types.FundraisingKeeper, stakingKeeper types.StakingKeeper,
```
Comment (lubtd) [inline x/participation/keeper/keeper.go:32]:
```suggestion
	fundraisingKeeper types.FundraisingKeeper,
	stakingKeeper types.StakingKeeper,
```

File: x/participation/module.go [hunk 7]
```diff
@@ -0,0 +1,171 @@
...
+
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/codec"
+	cdctypes "github.com/cosmos/cosmos-sdk/codec/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/module"
+
+	"github.com/tendermint/spn/x/participation/client/cli"
+	"github.com/tendermint/spn/x/participation/keeper"
+	"github.com/tendermint/spn/x/participation/types"
```
Comment (lubtd) [inline x/participation/module.go:23]: Need to goimport

File: x/participation/types/types.go [hunk 8]
```diff
@@ -0,0 +1 @@
+package types
```
Comment (lubtd) [inline x/participation/types/types.go:1]:
```suggestion
```
Let's remove this file, I don't know why we have it in Starport template

**Review by lubtd — COMMENTED** [review]
Great to have placed the boilerplates for the params already👍

File: x/participation/module.go (hunk 7 above)
Comment (giunatale) [reply x/participation/module.go:23]: goimport wasn't doing anything so I ended up manually mirroring imports format of `launch` module

**Comment by giunatale** [comment]
@lubtd thanks for all the suggestions. @aljo242 @Pantani let't try to have this merged quickly as it's blocking further work on participation 🙏  please take a look whenever you have time

**Review by lubtd — DISMISSED** [review]
Not a blocker but if we can add default params in `genesis_template.json` for `participation` in `localnet` that would be great

**Comment by giunatale** [comment]
We'll have to edit them when we have proper defaults, but no problem

**Comment by giunatale** [comment]
> add default params in `genesis_template.json` for `participation`

@lubtd LMK if values are reasonable placeholders. feel free to suggest changes

File: x/participation/simulation/simap.go [hunk 9]
```diff
@@ -0,0 +1,15 @@
+package simulation
```
Comment (aljo242) [inline x/participation/simulation/simap.go:1]: rename file to `simapp.go` later at some point

**Review by aljo242 — APPROVED** [review]

File: x/participation/simulation/simap.go (hunk 9 above)
Comment (giunatale) [reply x/participation/simulation/simap.go:1]: can do now if needed. you'll have to reapprove though

Comment (giunatale) [reply x/participation/simulation/simap.go]: @aljo242

**Review by aljo242 — APPROVED** [review]

**Review by lubtd — APPROVED** [review]
🚀

**Pull request merged**